### DISCUSSION TITLE
Enterprise security and module federation

### DISCUSSION THREAD

**Discussion by blackbaud-jeremymorgan** [webpack/webpack#14231] [Q&A] [answered]
I want to introduce module federation to my company.

Right now, my company has an ecosystem set up where multiple Angular SPAs are deployed under a common domain. If SPA A wants to utilize another SPA's (SPA B) modal for whatever, the system in place relies on opening a browser-sized iframe to SPA B where the modal is displayed.

This is the system I want module federation to replace, but I do have a big question - is there a way to prevent 3rd party applications from loading modals they shouldn't? If webpack is too low-level like I fear it might be, what is a common pattern people are doing to accomplish permission checking?

**Comment by alexander-akait** [comment, marked as answer]
Without module federation your clients also can load your code, no security problems here, anyway if you want to do it more security, you should setup this on server level

**Reply by blackbaud-jeremymorgan** [reply]
Ohhh good point, that clients can load the app bundle as-is now.
